v2/yarpctchannel/outbound.go

```diff
@@ -163,9 +164,15 @@ func callWithPeer(ctx context.Context, req *yarpc.Request, reqBody *yarpc.Buffer
 			"does not match the service name received in the response: sent %q, got: %q", req.Service, resSvcName)
 	}
 
+	var appErr error
+	if res.ApplicationError() {
+		// TODO(mhp): this is a filler error for now to preserve current yarpc
+		// error behavior. This should later be more well-defined.
+		appErr = errors.New("application error")
```

Comment:
We can instead use the same header grabbed from `getResponseErrorAndDeleteHeaderKeys(..)` right?

Reply:
I've tried this, but it looks like it fails a test when I do this. It looks like.. even if there was an application error, we don't actually store it in the headers...? So `getResponseErrorAndDeleteHeaderKeys(..)` will return `nil`, even though we want `response.ApplicationError` to be non-nil. I'm not exactly sure what's going on there.

I'm going to leave as is, considering that this diff is entirely preserving behavior. I'll leave the TODO and revisit it.

Reply:
Ahh yes. This is TChannel so the error is in the body. The error returned there would be the `yarpcerror.Status` retrieved from headers.

We'll fix this in the follow up regardless.